### ISSUE TITLE
Can transformer be a generator?

### ISSUE THREAD

**Issue by fuweifu-vtoo** [williamSYSU/TextGAN-PyTorch#41]
I notice that generator the porject use is only LSTMGenerator, i wonder if a basic transformer can be a generator?
Hope your response, thanks!

**Comment by sunahmet** [comment]
Yes, it's possible. Tried in this paper and showed better results: https://www.aclweb.org/anthology/N18-1122.pdf also it has implementation with TensorFlow: https://github.com/ZhenYangIACAS/NMT_GAN
